### PR TITLE
Refactor/apis

### PR DESCRIPTION
**Pull request by lm-sousa**
- [x] **[WARNING] Merging of this PR is dependant on the prior merge of specs-feup/lara-framework#45**
- [x] **[WARNING] Merging of this PR is dependant on the prior merge of specs-feup/clava#133**

### CODE REVIEW

File: Clava-JS/src-api/clava/pass/SingleReturnFunction.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  protected _apply_impl($jp: Joinpoint): PassResult {
+
+    if (!($jp instanceof FunctionJp) || !$jp.isImplementation) {
+      return this.new_result($jp, false);
+    }
+    const $body = $jp.body;
+    const $returnStmts = Query.searchFrom($body, "returnStmt").get() as ReturnStmt[];
+    if (
+      $returnStmts.length === 0 ||
+      ($returnStmts.length === 1 && $body.lastChild.instanceOf("returnStmt"))
```
Comment (joaobispo) [inline Clava-JS/src-api/clava/pass/SingleReturnFunction.ts:27]: instanceof

File: ClavaLaraApi/src-lara/clava/clava/analysis/CheckBasedAnalyser.js [hunk 2]
```diff
@@ -29,9 +29,10 @@ export default class CheckBasedAnalyser extends Analyser {
      */
     analyse($startNode = Query.root()) {
         const checkResultList = [];
-        for (const $node of Query.searchFrom($startNode)) {
+        for (const a of Query.searchFrom($startNode)) {
+            const $jp = a;
```
Comment (joaobispo) [inline ClavaLaraApi/src-lara/clava/clava/analysis/CheckBasedAnalyser.js:33]: Unnecessary?

File: ClavaWeaver/src/pt/up/fe/specs/clava/weaver/CxxWeaver.java [hunk 3]
```diff
@@ -2079,7 +2079,7 @@ public List<LaraResourceProvider> getNpmResources() {
         return Stream.concat(
                 super.getNpmResources().stream(),
                 Arrays.asList(ClavaApiJsResource.values()).stream())
-            .toList();
+            .collect(Collectors.toList());
```
Comment (joaobispo) [inline ClavaWeaver/src/pt/up/fe/specs/clava/weaver/CxxWeaver.java:2082]: Revert, .toList() is more recent

File: ClavaWeaver/src/pt/up/fe/specs/clava/weaver/joinpoints/CxxOmp.java [hunk 4]
```diff
@@ -239,8 +239,8 @@ public void setScheduleChunkSizeImpl(String chunkSize) {
     }
 
     @Override
-    public void setScheduleChunkSizeImpl(Integer chunkSize) {
-        setScheduleChunkSize(chunkSize.toString());
+    public void setScheduleChunkSizeImpl(int chunkSize) {
+        setScheduleChunkSize(String.valueOf(chunkSize));
```
Comment (joaobispo) [inline ClavaWeaver/src/pt/up/fe/specs/clava/weaver/joinpoints/CxxOmp.java:243]: use Integer.toString()

File: ClavaLaraApi/src-lara/clava/clava/analysis/CheckBasedAnalyser.js (hunk 2 above)
Comment (lm-sousa) [reply ClavaLaraApi/src-lara/clava/clava/analysis/CheckBasedAnalyser.js:33]: In the original Typescript file, the code appears as:
```typescript
const $jp = a as Joinpoint;
```
When transpiled, the type information is removed.

File: Clava-JS/src-api/clava/pass/SingleReturnFunction.ts (hunk 1 above)
Comment (lm-sousa) [reply Clava-JS/src-api/clava/pass/SingleReturnFunction.ts:27]: Resolved on a later PR.

File: ClavaWeaver/src/pt/up/fe/specs/clava/weaver/CxxWeaver.java (hunk 3 above)
Comment (lm-sousa) [reply ClavaWeaver/src/pt/up/fe/specs/clava/weaver/CxxWeaver.java:2082]: At the time, we were targeting Java 14. Now that Java 17 is the default we can use that method directly.

Resolved in bbb65de65

File: ClavaWeaver/src/pt/up/fe/specs/clava/weaver/joinpoints/CxxOmp.java (hunk 4 above)
Comment (lm-sousa) [reply ClavaWeaver/src/pt/up/fe/specs/clava/weaver/joinpoints/CxxOmp.java:243]: Resolved in bbb65de65

**Pull request merged**